### ISSUE TITLE
WSL2: 'code .' resulting in Segmentation fault

### ISSUE THREAD

**Issue by SHJordan** [microsoft/vscode-remote-release#3117]
Issue Type: <b>Bug</b>

navigate to desired folder to be opened by wsl2 ubuntu 20.04, write "code ." and hit enter.

VS Code version: Code 1.45.1 (5763d909d5f12fe19f215cbfdd29a91c0fa9208a, 2020-05-14T08:27:35.169Z)
OS version: Windows_NT x64 10.0.19635

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i5-6600K CPU @ 3.50GHz (4 x 3504)|
|GPU Status|2d_canvas: enabled<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>oop_rasterization: disabled_off<br>protected_video_decode: unavailable_off<br>rasterization: enabled<br>skia_renderer: disabled_off_ok<br>video_decode: enabled<br>viz_display_compositor: enabled_on<br>viz_hit_test_surface_layer: disabled_off_ok<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|15.96GB (5.56GB free)|
|Process Argv||
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (14)</summary>

Extension|Author (truncated)|Version
---|---|---
copy-relative-path|ale|0.0.2
dart-code|Dar|3.11.0
flutter|Dar|3.11.0
vscode-eslint|dba|2.1.5
prettier-vscode|esb|5.0.0
intellij-idea-keybindings|k--|0.2.40
expand-region|let|0.1.4
vscode-edge-devtools|ms-|1.0.7
remote-wsl|ms-|0.44.2
debugger-for-edge|msj|1.0.14
vetur|oct|0.24.0
material-icon-theme|PKi|4.1.0
vue-vscode-snippets|sdr|1.9.0
vscode-wakatime|Wak|4.0.0

(2 theme extensions excluded)

</details>

**Comment by aeschli** [comment]
Can you run `bash -x $(which code)` to get more information?
Also `code --verbose`

**Comment by SHJordan** [comment]
> code --verbose

![image](https://user-images.githubusercontent.com/884602/83669310-bac7d000-a5a7-11ea-8417-f9fd66c78222.png)
Sadly it seems not to be working at all.

**Comment by SHJordan** [comment]
`which code` worked for me:
/mnt/c/Program Files/Microsoft VS Code/bin/code

**Comment by aeschli** [comment]
Sorry, forgot the quotes:
`bash -x "$(which code)"`

**Comment by SHJordan** [comment]
> Sorry, forgot the quotes:
> `bash -x "$(which code)"`

+ COMMIT=5763d909d5f12fe19f215cbfdd29a91c0fa9208a
+ APP_NAME=code
+ QUALITY=stable
+ NAME=Code
+ DATAFOLDER=.vscode
++++ realpath '/mnt/c/Program Files/Microsoft VS Code/bin/code'
+++ dirname '/mnt/c/Program Files/Microsoft VS Code/bin/code'
++ dirname '/mnt/c/Program Files/Microsoft VS Code/bin'
+ VSCODE_PATH='/mnt/c/Program Files/Microsoft VS Code'
+ ELECTRON='/mnt/c/Program Files/Microsoft VS Code/Code.exe'
+ grep -qi Microsoft /proc/version
++ uname -r
++ sed -E 's/^[0-9.]+-([0-9]+)-Microsoft|([0-9]+).([0-9]+).([0-9]+)-microsoft-standard|.*/\1\2\3\4/'
+ WSL_BUILD=419104
+ '[' -z 419104 ']'
+ '[' 419104 -ge 17063 ']'
+ export WSLENV=ELECTRON_RUN_AS_NODE/w:WT_SESSION::WT_PROFILE_ID
+ WSLENV=ELECTRON_RUN_AS_NODE/w:WT_SESSION::WT_PROFILE_ID
++ wslpath -m '/mnt/c/Program Files/Microsoft VS Code/resources/app/out/cli.js'
+ CLI='C:/Program Files/Microsoft VS Code/resources/app/out/cli.js'
+ WSL_EXT_ID=ms-vscode-remote.remote-wsl
+ '[' 419104 -ge 41955 -a 419104 -lt 41959 ']'
+ ELECTRON_RUN_AS_NODE=1
+ '/mnt/c/Program Files/Microsoft VS Code/Code.exe' 'C:/Program Files/Microsoft VS Code/resources/app/out/cli.js' --locate-extension ms-vscode-remote.remote-wsl
++ cat /tmp/remote-wsl-loc.txt
+ WSL_EXT_WLOC='c:\Users\Jordan\.vscode\extensions\ms-vscode-remote.remote-wsl-0.44.2'
+ '[' -n 'c:\Users\Jordan\.vscode\extensions\ms-vscode-remote.remote-wsl-0.44.2' ']'
++ wslpath -u 'c:\Users\Jordan\.vscode\extensions\ms-vscode-remote.remote-wsl-0.44.2'
+ WSL_CODE=/mnt/c/Users/Jordan/.vscode/extensions/ms-vscode-remote.remote-wsl-0.44.2/scripts/wslCode.sh
+ /mnt/c/Users/Jordan/.vscode/extensions/ms-vscode-remote.remote-wsl-0.44.2/scripts/wslCode.sh 5763d909d5f12fe19f215cbfdd29a91c0fa9208a stable '/mnt/c/Program Files/Microsoft VS Code/Code.exe' code .vscode
Segmentation fault
+ exit 139

**Comment by derkan** [comment]
same error here, any improvements?

**Comment by Darival** [comment]
same issue here:
WSL2
Ubuntu 20.04 focal(on the Windows Subsystem for Linux)

```bash
$ bash -x "$(which code)"
+ '[' '' = true ']'
+ COMMIT=17299e413d5590b14ab0340ea477cdd86ff13daf
+ APP_NAME=code
+ QUALITY=stable
+ NAME=Code
+ DATAFOLDER=.vscode
++++ realpath '/mnt/c/Users/cesar/AppData/Local/Programs/Microsoft VS Code/bin/code'
+++ dirname '/mnt/c/Users/cesar/AppData/Local/Programs/Microsoft VS Code/bin/code'
++ dirname '/mnt/c/Users/cesar/AppData/Local/Programs/Microsoft VS Code/bin'
+ VSCODE_PATH='/mnt/c/Users/cesar/AppData/Local/Programs/Microsoft VS Code'
+ ELECTRON='/mnt/c/Users/cesar/AppData/Local/Programs/Microsoft VS Code/Code.exe'
+ IN_WSL=false
+ '[' -n Ubuntu ']'
+ IN_WSL=true
+ '[' true = true ']'
+ export WSLENV=ELECTRON_RUN_AS_NODE/w:WT_SESSION::WT_PROFILE_ID
+ WSLENV=ELECTRON_RUN_AS_NODE/w:WT_SESSION::WT_PROFILE_ID
++ wslpath -m '/mnt/c/Users/cesar/AppData/Local/Programs/Microsoft VS Code/resources/app/out/cli.js'
+ CLI='C:/Users/cesar/AppData/Local/Programs/Microsoft VS Code/resources/app/out/cli.js'
+ WSL_EXT_ID=ms-vscode-remote.remote-wsl
+ ELECTRON_RUN_AS_NODE=1
+ '/mnt/c/Users/cesar/AppData/Local/Programs/Microsoft VS Code/Code.exe' 'C:/Users/cesar/AppData/Local/Programs/Microsoft VS Code/resources/app/out/cli.js' --locate-extension ms-vscode-remote.remote-wsl
++ cat /tmp/remote-wsl-loc.txt
+ WSL_EXT_WLOC='c:\Users\cesar\.vscode\extensions\ms-vscode-remote.remote-wsl-0.44.4'
+ '[' -n 'c:\Users\cesar\.vscode\extensions\ms-vscode-remote.remote-wsl-0.44.4' ']'
++ wslpath -u 'c:\Users\cesar\.vscode\extensions\ms-vscode-remote.remote-wsl-0.44.4'
+ WSL_CODE=/mnt/c/Users/cesar/.vscode/extensions/ms-vscode-remote.remote-wsl-0.44.4/scripts/wslCode.sh
+ /mnt/c/Users/cesar/.vscode/extensions/ms-vscode-remote.remote-wsl-0.44.4/scripts/wslCode.sh 17299e413d5590b14ab0340ea477cdd86ff13daf stable '/mnt/c/Users/cesar/AppData/Local/Programs/Microsoft VS Code/Code.exe' code .vscode
Segmentation fault
+ exit 139
```
```bash
$code --verbose
Segmentation fault
```

**Comment by Edmund-Shi** [comment]
I got the same issue in WSL 2, Ubuntu 18.04. After I switch WSL 2 to WSL 2 using `wsl --set-version Ubuntu 1`, I still got the same issue.

<details>
<summary>Output for <code>bash -x "$(which code)"</code></summary>
<pre><code>
+ '[' '' = true ']'
+ COMMIT=db40434f562994116e5b21c24015a2e40b2504e6
+ APP_NAME=code
+ QUALITY=stable
+ NAME=Code
+ DATAFOLDER=.vscode
++++ realpath '/mnt/c/Users/edm/AppData/Local/Programs/Microsoft VS Code/bin/code'
+++ dirname '/mnt/c/Users/edm/AppData/Local/Programs/Microsoft VS Code/bin/code'
++ dirname '/mnt/c/Users/edm/AppData/Local/Programs/Microsoft VS Code/bin'
+ VSCODE_PATH='/mnt/c/Users/edm/AppData/Local/Programs/Microsoft VS Code'
+ ELECTRON='/mnt/c/Users/edm/AppData/Local/Programs/Microsoft VS Code/Code.exe'
+ IN_WSL=false
+ '[' -n Ubuntu ']'
+ IN_WSL=true
+ '[' true = true ']'
+ export WSLENV=ELECTRON_RUN_AS_NODE/w:WT_SESSION::WT_PROFILE_ID
+ WSLENV=ELECTRON_RUN_AS_NODE/w:WT_SESSION::WT_PROFILE_ID
++ wslpath -m '/mnt/c/Users/edm/AppData/Local/Programs/Microsoft VS Code/resources/app/out/cli.js'
+ CLI='C:/Users/edm/AppData/Local/Programs/Microsoft VS Code/resources/app/out/cli.js'
+ WSL_EXT_ID=ms-vscode-remote.remote-wsl
+ ELECTRON_RUN_AS_NODE=1
+ '/mnt/c/Users/edm/AppData/Local/Programs/Microsoft VS Code/Code.exe' 'C:/Users/edm/AppData/Local/Programs/Microsoft VS Code/resources/app/out/cli.js' --locate-extension ms-vscode-remote.remote-wsl
++ cat /tmp/remote-wsl-loc.txt
+ WSL_EXT_WLOC='c:\Users\edm\.vscode\extensions\ms-vscode-remote.remote-wsl-0.44.4'
+ '[' -n 'c:\Users\edm\.vscode\extensions\ms-vscode-remote.remote-wsl-0.44.4' ']'
++ wslpath -u 'c:\Users\edm\.vscode\extensions\ms-vscode-remote.remote-wsl-0.44.4'
+ WSL_CODE=/mnt/c/Users/edm/.vscode/extensions/ms-vscode-remote.remote-wsl-0.44.4/scripts/wslCode.sh
+ /mnt/c/Users/edm/.vscode/extensions/ms-vscode-remote.remote-wsl-0.44.4/scripts/wslCode.sh db40434f562994116e5b21c24015a2e40b2504e6 stable '/mnt/c/Users/edm/AppData/Local/Programs/Microsoft VS Code/Code.exe' code .vscode
Segmentation fault (core dumped)
+ exit 139
</code></pre>
</details>

<details>
<summary>Output for <code>wsl --list --verbose</code></summary>
<pre><code>
  NAME      STATE           VERSION
* Ubuntu    Running         1
</code></pre>
</details>

**Comment by drewhemm** [comment]
Woke up this morning to start work after latest Windows Insider Preview updates and now I am getting this! Is there any known workaround?

**Comment by drewhemm** [comment]
<details>
  <summary>Output for <code>bash -x "$(which code)"</code></summary>
  
```
+ '[' '' = true ']'
+ COMMIT=db40434f562994116e5b21c24015a2e40b2504e6
+ APP_NAME=code
+ QUALITY=stable
+ NAME=Code
+ DATAFOLDER=.vscode
++++ realpath '/mnt/c/Users/andre/AppData/Local/Programs/Microsoft VS Code/bin/code'
+++ dirname '/mnt/c/Users/andre/AppData/Local/Programs/Microsoft VS Code/bin/code'
++ dirname '/mnt/c/Users/andre/AppData/Local/Programs/Microsoft VS Code/bin'
+ VSCODE_PATH='/mnt/c/Users/andre/AppData/Local/Programs/Microsoft VS Code'
+ ELECTRON='/mnt/c/Users/andre/AppData/Local/Programs/Microsoft VS Code/Code.exe'
+ IN_WSL=false
+ '[' -n Ubuntu ']'
+ IN_WSL=true
+ '[' true = true ']'
+ export WSLENV=ELECTRON_RUN_AS_NODE/w:WT_SESSION::WT_PROFILE_ID
+ WSLENV=ELECTRON_RUN_AS_NODE/w:WT_SESSION::WT_PROFILE_ID
++ wslpath -m '/mnt/c/Users/andre/AppData/Local/Programs/Microsoft VS Code/resources/app/out/cli.js'
+ CLI='C:/Users/andre/AppData/Local/Programs/Microsoft VS Code/resources/app/out/cli.js'
+ WSL_EXT_ID=ms-vscode-remote.remote-wsl
+ ELECTRON_RUN_AS_NODE=1
+ '/mnt/c/Users/andre/AppData/Local/Programs/Microsoft VS Code/Code.exe' 'C:/Users/andre/AppData/Local/Programs/Microsoft VS Code/resources/app/out/cli.js' --locate-extension ms-vscode-remote.remote-wsl
++ cat /tmp/remote-wsl-loc.txt
+ WSL_EXT_WLOC='c:\Users\andre\.vscode\extensions\ms-vscode-remote.remote-wsl-0.44.4'
+ '[' -n 'c:\Users\andre\.vscode\extensions\ms-vscode-remote.remote-wsl-0.44.4' ']'
++ wslpath -u 'c:\Users\andre\.vscode\extensions\ms-vscode-remote.remote-wsl-0.44.4'
+ WSL_CODE=/mnt/c/Users/andre/.vscode/extensions/ms-vscode-remote.remote-wsl-0.44.4/scripts/wslCode.sh
+ /mnt/c/Users/andre/.vscode/extensions/ms-vscode-remote.remote-wsl-0.44.4/scripts/wslCode.sh db40434f562994116e5b21c24015a2e40b2504e6 stable '/mnt/c/Users/andre/AppData/Local/Programs/Microsoft VS Code/Code.exe' code .vscode
Segmentation fault
+ exit 139
```
</details>

<details>
  <summary>Output for <code>wsl --list --verbose</code></summary>
  
```
NAME                   STATE           VERSION
* Ubuntu                 Running         2
```
</details>

**Comment by aeschli** [comment]
Please also try `VSCODE_WSL_DEBUG_INFO=true code .`. This should get even more logging.

**Comment by drewhemm** [comment]
<details>
  <summary>Output for <code>VSCODE_WSL_DEBUG_INFO=true code .</code></summary>
  
```
+ COMMIT=db40434f562994116e5b21c24015a2e40b2504e6
+ APP_NAME=code
+ QUALITY=stable
+ NAME=Code
+ DATAFOLDER=.vscode
+ realpath /mnt/c/Users/andre/AppData/Local/Programs/Microsoft VS Code/bin/code
+ dirname /mnt/c/Users/andre/AppData/Local/Programs/Microsoft VS Code/bin/code
+ dirname /mnt/c/Users/andre/AppData/Local/Programs/Microsoft VS Code/bin
+ VSCODE_PATH=/mnt/c/Users/andre/AppData/Local/Programs/Microsoft VS Code
+ ELECTRON=/mnt/c/Users/andre/AppData/Local/Programs/Microsoft VS Code/Code.exe
+ IN_WSL=false
+ [ -n Ubuntu ]
+ IN_WSL=true
+ [ true = true ]
+ export WSLENV=ELECTRON_RUN_AS_NODE/w:WT_SESSION::WT_PROFILE_ID
+ wslpath -m /mnt/c/Users/andre/AppData/Local/Programs/Microsoft VS Code/resources/app/out/cli.js
+ CLI=C:/Users/andre/AppData/Local/Programs/Microsoft VS Code/resources/app/out/cli.js
+ WSL_EXT_ID=ms-vscode-remote.remote-wsl
+ ELECTRON_RUN_AS_NODE=1 /mnt/c/Users/andre/AppData/Local/Programs/Microsoft VS Code/Code.exe C:/Users/andre/AppData/Local/Programs/Microsoft VS Code/resources/app/out/cli.js --locate-extension ms-vscode-remote.remote-wsl
+ cat /tmp/remote-wsl-loc.txt
+ WSL_EXT_WLOC=c:\Users\andre\.vscode\extensions\ms-vscode-remote.remote-wsl-0.44.4
+ [ -n c:\Users\andre\.vscode\extensions\ms-vscode-remote.remote-wsl-0.44.4 ]
+ wslpath -u c:\Users\andre\.vscode\extensions\ms-vscode-remote.remote-wsl-0.44.4
+ WSL_CODE=/mnt/c/Users/andre/.vscode/extensions/ms-vscode-remote.remote-wsl-0.44.4/scripts/wslCode.sh
+ /mnt/c/Users/andre/.vscode/extensions/ms-vscode-remote.remote-wsl-0.44.4/scripts/wslCode.sh db40434f562994116e5b21c24015a2e40b2504e6 stable /mnt/c/Users/andre/AppData/Local/Programs/Microsoft VS Code/Code.exe code .vscode .
+ [ -z .vscode ]
+ echo .vscode+
sed -e s/.vscode/.vscode-server/g
+ DATAFOLDER=.vscode-server
+ VSCODE_REMOTE_BIN=/home/andrew/.vscode-server/bin
+ AUTHORITY=wsl+default
+ [ Ubuntu ]
+ AUTHORITY=wsl+Ubuntu
+ [ ! -d /home/andrew/.vscode-server ]
+ dirname /mnt/c/Users/andre/.vscode/extensions/ms-vscode-remote.remote-wsl-0.44.4/scripts/wslCode.sh
+ /mnt/c/Users/andre/.vscode/extensions/ms-vscode-remote.remote-wsl-0.44.4/scripts/wslDownload.sh db40434f562994116e5b21c24015a2e40b2504e6 stable /home/andrew/.vscode-server/bin
+ [ ! -d /home/andrew/.vscode-server/bin/db40434f562994116e5b21c24015a2e40b2504e6 ]
+ RC=0
+ [ 0 -ne 0 ]
+ dirname /mnt/c/Users/andre/.vscode/extensions/ms-vscode-remote.remote-wsl-0.44.4/scripts/wslCode.sh
+ VSCODE_CLIENT_COMMAND=/mnt/c/Users/andre/AppData/Local/Programs/Microsoft VS Code/Code.exe VSCODE_CLIENT_COMMAND_CWD=/mnt/c/Users/andre/.vscode/extensions/ms-vscode-remote.remote-wsl-0.44.4/scripts VSCODE_CLI_AUTHORITY=wsl+Ubuntu /home/andrew/.vscode-server/bin/db40434f562994116e5b21c24015a2e40b2504e6/bin/code .
Segmentation fault
+ exit 139
```
</details>

**Comment by aeschli** [comment]
The segmentation fault occurs when running the windows exe of code (/mnt/c/Users/andre/AppData/Local/Programs/Microsoft VS Code/Code.exe)

Must be a WSL issue. Maybe related to https://github.com/microsoft/WSL/issues/5274
@drewhemm You can try WSLENV="" VSCODE_WSL_DEBUG_INFO=true code .

**Comment by drewhemm** [comment]
I solved this by renaming the `~/.vscode-server` directory to `~/.vscode-server.bak` and then re-opening VSC. I also tried a few things like re-creating my WSL instance, but as soon as I copied the contents of my home directory over, I got the segmentation fault again. 

I have copied my extensions over from the .bak folder, so it is something else that is causing the issue.

**Comment by drewhemm** [comment]
I also used the 'opportunity' to move over to the newer Ubuntu 20.04 LTS WSL distribution. I used a regex (`apt install ([\w-]+)` on `/var/log/apt/history.log`) to get the names of all the apt packages I had installed on my previous instance and then install them all in one go. Everything else I need is in my home folder (like pip packages installed with `pip install --user ansible`), so the transition was quite seamless.

**Comment by Edmund-Shi** [comment]
> I solved this by renaming the `~/.vscode-server` directory to `~/.vscode-server.bak` and then re-opening VSC. 

Thanks for @drewhemm. This works for me!

**Comment by derkan** [comment]
> The segmentation fault occurs when running the windows exe of code (/mnt/c/Users/andre/AppData/Local/Programs/Microsoft VS Code/Code.exe)
> 
> Must be a WSL issue. Maybe related to [microsoft/WSL#5274](https://github.com/microsoft/WSL/issues/5274)
> @drewhemm You can try WSLENV="" VSCODE_WSL_DEBUG_INFO=true code .

this solution(`WSLENV=""`) worked for me, thanks.

**Comment by aeschli** [comment]
We have to issues on segmentation faults
a. when opening VSCode with `code .`
b. when VSCode is open but starting the remote server fails

This issue is for a. 
It seems to be a WSL issue related to WSLENV that should be fixed soon: microsoft/WSL#5274
In the meantime set your WSLENV to empty or run with  ` WSLENV="" code .`

For b. we have https://github.com/microsoft/vscode-remote-release/issues/3556

Closing this issue.
